### PR TITLE
[UNI-257] updated ModelService and tests

### PR DESCRIPTION
**Pull request by marionleborgne**
First round for UNI-257:
* Updated Model Service
* Updated tests
* Spawning `model_runner_2.py` with the right params (`unicorn/tests/js/unit/fixtures`)

cc @lscheinkman @brev  if you can have a look, that'd be great (disregard `ParamFinderService.js` for now - working on it)

> Note: don't merge yet. Now adding the Param Finder Service (last part of UNI-257)

### CODE REVIEW

**Comment by brev** [comment]
@marionleborgne Looks like a great start.

The test for `ModelService2` is failing for me, I seem to be missing `fixtures/rec-center.csv` (I have `fixtures/rec-center-15.csv`, but that doesn't work).

Otherwise :+1: so far. Seem to be going ok from your side?

**Comment by marionleborgne** [comment]
@brev thanks, I added the missing CSV file. This should work now. Making good progress with the ParamFinderService.

**Comment by brev** [comment]
@marionleborgne tests passing for me also, thanks

**Comment by marionleborgne** [comment]
Added param finder service and tests. cc @lscheinkman & @brev

File: unicorn/js/main/ParamFinderService.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+   */
+  getParamFinder() {
+    return this._paramFinder;
+  }
+
+  /**
+   * Stops param finder .
+   */
+  stopParamFinder() {
+    this._paramFinder.child.kill();
```
Comment (lscheinkman) [inline unicorn/js/main/ParamFinderService.js]: `this._paramFinder` may be `null` if the child process was closed before `stopParamFinder` was called.

File: unicorn/js/main/ParamFinderService.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+ *
+ * http://numenta.org/licenses/
+ * -------------------------------------------------------------------------- */
+
+import childProcess from 'child_process';
+import EventEmitter from 'events';
+import path from 'path';
+import UserError from './UserError';
+
+export const PARAM_FINDER_EVENT_TYPE = 'PARAM_FINDER_EVENT_TYPE';
```
Comment (lscheinkman) [inline unicorn/js/main/ParamFinderService.js]: I'm not sure this is necessary. You can just call `emit` without this extra value.
See https://nodejs.org/api/events.html#events_emitter_emit_event_arg1_arg2
This way the event handling code would look like this:
```
     child.on('error', (error) => {
       this.emit('error', error);
     });
 
     child.stderr.on('data', (error) => {
       this.emit('error', error);
     });
 
     child.stdout.on('data', (data) => {
       this.emit('data', data);
     });
 
     child.once('close', (code) => {
       this._paramFinder = null;
       this.emit('close', code);
     });
```

File: unicorn/js/main/ParamFinderService.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    return this._paramFinder;
+  }
+
+  /**
+   * Stops param finder .
+   */
+  stopParamFinder() {
+    this._paramFinder.child.kill();
+    this._paramFinder = null;
+    this.removeAllListeners(PARAM_FINDER_EVENT_TYPE);
```
Comment (lscheinkman) [inline unicorn/js/main/ParamFinderService.js]: Can be replaces with `this.removeAllListeners()`

File: unicorn/js/main/ParamFinderServiceIPC.js [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+ *
+ * You should have received a copy of the GNU Affero Public License along with
+ * this program. If not, see http://www.gnu.org/licenses.
+ *
+ * http://numenta.org/licenses/
+ * -------------------------------------------------------------------------- */
+
+import {ipcMain as ipc} from 'electron';
+
+import {ParamFinderService, PARAM_FINDER_EVENT_TYPE} from './ParamFinderService';
```
Comment (lscheinkman) [inline unicorn/js/main/ParamFinderServiceIPC.js]: `PARAM_FINDER_EVENT_TYPE` can be removed

File: unicorn/js/main/PortablePython.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  }
+}
+
+/**
+ * Get path to platform specific portable python distribution.
+ * See 'optionalDependencies' in 'package.json' for available platforms.
+ * @throws {@link PortablePythonNotFoundError}
+ * @returns {String} Path to portable python distribution
+ */
+export default function getPortablePython() {
```
Comment (lscheinkman) [inline unicorn/js/main/PortablePython.js]: :+1: Great idea!

**Comment by lscheinkman** [comment]
:+1:  LGTM

**Pull request merged**